### PR TITLE
fix: l0RowCount metrics value always empty

### PR DESCRIPTION
**Pull request by XuanYang-cn**
See also: #36953

### CODE REVIEW

File: internal/datacoord/segment_info.go [hunk 1]
```diff
@@ -508,9 +508,10 @@ func (s *SegmentInfo) getSegmentSize() int64 {
 	return s.size.Load()
 }
 
-// getDeltaCount use cached value when segment is immutable
+// L1 segment deltaCount changes in any state
+// L0 segment deltaCount won't change
 func (s *SegmentInfo) getDeltaCount() int64 {
-	if s.deltaRowcount.Load() < 0 || s.State != commonpb.SegmentState_Flushed {
+	if s.deltaRowcount.Load() <= 0 || s.GetLevel() != datapb.SegmentLevel_L0 {
```
Comment (congqixia) [inline internal/datacoord/segment_info.go:514]: Make make `getDeltaCount` calculate every call for L0 segment. Looks like not by design?

Comment (XuanYang-cn) [reply internal/datacoord/segment_info.go:514]: this will calculate each time for No L0 segment

Comment (congqixia) [reply internal/datacoord/segment_info.go:514]: For flushed segment, it shall be calculated only once.

Comment (XuanYang-cn) [reply internal/datacoord/segment_info.go:514]: But flushed segments delete rowcounts changes all the time. Besides this functions will be called by L0 segments only.

**Comment by XuanYang-cn** [comment]
rerun ut

**Comment by XuanYang-cn** [comment]
rerun go-sdk

**Comment by XuanYang-cn** [comment]
rerun go-sdk

**Comment by XuanYang-cn** [comment]
rerun go-sdk

**Comment by XuanYang-cn** [comment]
rerun ut

**Comment by XuanYang-cn** [comment]
rerun go-sdk

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**